test/client/metadataApiDeploy.test.ts

```diff
@@ -548,13 +548,13 @@ describe('MetadataApiDeploy', () => {
             fullName,
             type: type.name,
             state: ComponentStatus.Changed,
-            filePath: content,
+            filePath: ensureString(content),
```

Comment:
there's a bunch of these in the test.  The mocks come from SourceComponent, which has `xml` and `content` as optional properties.

the tests assert that they're there.

It's harder to work with the SourceComponent class (ex: to make SourceComponentWithXmlAndContent, etc) the way you would a simple object/type or to put a type guard around it  
...
so it's probably simpler to do this `ensure` stuff for UT.